Add Python Solution to CSES Playlist (#3874)
cpinitiative/usaco-guide: content/4_Gold/Sliding_Window.mdx

</CPPSection>

<PySection>

```py
def main():

> 	"""This code is put in a function for performance reasons."""

Reviewer: does it tle outside of main?

Author: yeah

	n = int(input().strip())
	nums = [int(v) for v in input().split(" ")]

	ans = -float("inf")
	left, right = 0, 0
	unique_songs = set()
	while right < n:

> 		# Invariant: all the songs in `unique_songs` are unique in each iteration.
> 		# To maintain this invariant, shrink the window before inserting nums[right].

Reviewer: ```suggestion
		# Notice that all the songs in `unique_songs` are unique in each iteration.
		# We keep this property by shrinking the window before inserting nums[right].
```

Author: Done!
